src/fairchem/data/oc/core/adsorbate.py

```diff
@@ -62,27 +65,29 @@ def __init__(
                 )
             else:
                 self.binding_indices = adsorbate_binding_indices
-        elif adsorbate_id_from_db is not None:
-            adsorbate_db = adsorbate_db or pickle.load(open(adsorbate_db_path, "rb"))
-            self._load_adsorbate(adsorbate_db[adsorbate_id_from_db])
-        elif adsorbate_smiles_from_db is not None:
-            adsorbate_db = adsorbate_db or pickle.load(open(adsorbate_db_path, "rb"))
-            adsorbate_obj_tuple = [
-                (idx, adsorbate_info)
-                for idx, adsorbate_info in adsorbate_db.items()
-                if adsorbate_info[1] == adsorbate_smiles_from_db
-            ]
-            if len(adsorbate_obj_tuple) < 1:
-                warnings.warn(
-                    "An adsorbate with that SMILES string was not found. Choosing one at random instead."
-                )
-                self._get_adsorbate_from_random(adsorbate_db)
-            else:
-                self._load_adsorbate(adsorbate_obj_tuple[0][1])
-                self.adsorbate_id_from_db = adsorbate_obj_tuple[0][0]
         else:
-            adsorbate_db = adsorbate_db or pickle.load(open(adsorbate_db_path, "rb"))
-            self._get_adsorbate_from_random(adsorbate_db)
+            if adsorbate_db is None:
+                with open(adsorbate_db_path, "rb") as fp:
+                    adsorbate_db = pickle.load(fp)
+
+            if adsorbate_id_from_db is not None:
+                self._load_adsorbate(adsorbate_db[adsorbate_id_from_db])
+            elif adsorbate_smiles_from_db is not None:
+                adsorbate_obj_tuple = [
+                    (idx, adsorbate_info)
+                    for idx, adsorbate_info in adsorbate_db.items()
+                    if adsorbate_info[1] == adsorbate_smiles_from_db
+                ]
+                if len(adsorbate_obj_tuple) < 1:
+                    warnings.warn(
+                        "An adsorbate with that SMILES string was not found. Choosing one at random instead."
+                    )
+                    self._get_adsorbate_from_random(adsorbate_db)
+                else:
+                    self._load_adsorbate(adsorbate_obj_tuple[0][1])
+                    self.adsorbate_id_from_db = adsorbate_obj_tuple[0][0]
+            else:
+                self._get_adsorbate_from_random(adsorbate_db)
```

Comment:
No way ruff was able to do this. That's impressive! Or did you manually resolve this?

Reply:
Nah, this one was done manually!